BlueprintUI/Sources/Internal/HintingSizeCache.swift

```diff
@@ -26,7 +31,59 @@ final class HintingSizeCache {
             return size
         }
 
-        // TODO: implement hinting
+        // This option allows us to deduce if the key is within a hinted range even if it does not
+        // land on a boundary value, by searching "well known" keys.
+        if options.searchUnconstrainedKeys {
+
+            // When there are no unconstrained keys, searching for them here can have a significant
+            // performance hit. Unconstrained keys are usually not present unless introduced by an
+            // element like a ScrollView.
```

Comment:
Oh nice! Glad you were able to find a solution for this; it's a huge resource suck right now.